File: 2021/1-2. Sonar Sweep/C++/src/main.cpp
#include <cstddef>
#include <fstream>
#include <iostream>
#include <numeric>
#include <vector>

int main()
{
    std::ifstream infile("input.txt");

    if (!infile.is_open())
    {
        std::cerr << "ERROR! Could not open input.txt!\n";
        return -1;
    }

    std::vector<int> data;

    int tmp;

    while (infile >> tmp)
    {
        data.emplace_back(tmp);
    }

    int increases = 0;

    for (size_t i = 0, j = 1; i < data.size() - (data.size() % 3); i++, j++)
    {
        const int sum1 = std::accumulate(data.begin() + static_cast<long int>(i), data.begin()  + static_cast<long int>(i) + 3, 0);
        const int sum2 = std::accumulate(data.begin() + static_cast<long int>(j), data.begin()  + static_cast<long int>(j) + 3, 0);

        if (sum2 > sum1)
            increases++;
    }

    std::cout << "Answer: " << increases << '\n';

    return 0;
}

File: 2021/1-2. Sonar Sweep/Python/main.py
with open("input.txt") as file:
    nums = [int(x) for x in file]

increases = 0

i = 0
j = 1

while i < len(nums) - len(nums) % 3:
    sum1 = sum(nums[i:i + 3])
    sum2 = sum(nums[j:j + 3])

    if sum2 > sum1:
        increases += 1

    i +=1
    j +=1

print("Answer:", increases)